src/movies.js

```diff
@@ -1,16 +1,56 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+console.log(movies.length);
+const getAllDirectors = (array) => {
+    const directores = array.map((e) => {
+        return e.director
+    })
+    return directores;
+}
+getAllDirectors(movies)
+console.log(getAllDirectors(movies));
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+const howManyMovies = (arr) => ((arr.filter(e => e.director === "Steven Spielberg")).filter(e => e.genre.includes("Drama"))).length;
+console.log(howManyMovies(movies));
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
+const ratesAverage = (array) => {
+    const promedios = array.reduce((acumulado, actual) => {
+        if (actual.rate !== null && actual.rate !== '' && typeof actual.rate !== "undefined") {
+            acumulado += actual.rate;
+        }
+        return acumulado;
+    }, 0)
+    if (promedios === 0) {
+        return 0
+    }
+    let puntuacion = (promedios / array.length)
+    return parseInt(puntuacion.toFixed(2))
 // Iteration 4: Drama movies - Get the average of Drama Movies
-
+const dramaMoviesRate = (arr) => {
+    let newArr = [];
+    arr.map(e => e.genre.includes('Drama') ? newArr.push(e) : newArr);
```

Comment:
Aquí me parece algo redundante hacer un variable con un array vacío y pasar le los elementos del map, ya que el metodo map devuelve por si mismo un array nuevo (Aquí lo usaron como si fuera un forEach sin embargo estaria bien que repasaran un poco como funcionan cada uno). 
Creo que filter nos ayudaría mas a obtener las películas de Drama 😉 
```suggestion
  const newArr = arr.filter (e => { return e.genre.includes('Drama') });
    console.log(newArr)
```